### PR TITLE
Add board specific hooks, flash all leds during bootloader mode

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

**Comment by nomis52** [comment]
Closed #152 and gets us a working Number8!

File: firmware/src/system_config/number8/board_init.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * board_init.h
```
Comment (peternewman) [inline firmware/src/system_config/number8/board_init.h]: Do you not want the parent folder here to be consisten?

File: firmware/src/usb_descriptors.c [hunk 2]
```diff
@@ -101,7 +102,7 @@ static const uint8_t g_config_descriptor[] = {
   1,  // Index value of this configuration
   0,  // Configuration string index
   USB_ATTRIBUTE_DEFAULT | USB_ATTRIBUTE_SELF_POWERED,  // Attributes
-  50,  // Max power consumption (2X mA)
+  USB_POWER_CONSUMPTION,  // Max power consumption (2X mA)
```
Comment (peternewman) [inline firmware/src/usb_descriptors.c]: This seems to contradict the comment above. I think you want mA/2 for the value here.

File: tests/system_config/bootloader_settings.h [hunk 3]
```diff
@@ -60,4 +63,17 @@ const PORTS_CHANNEL LED_PORT_CHANNEL = PORT_CHANNEL_D;
...
 
+/**
+ * @brief The LEDS to flash in bootloader mode.
+ */
+const Bootloader_LEDs BOOTLOADER_LEDS = {
+  .count = 1,
+  .leds = {
+    {
+      .port_channel = PORT_CHANNEL_D,
+      .port_bit = PORTS_BIT_POS_0
```
Comment (peternewman) [inline tests/system_config/bootloader_settings.h]: Can't you say LED_PORT_BIT given what you've defined above?

**Comment by peternewman** [comment]
Generally LGTM

File: firmware/src/system_config/number8/board_init.h (hunk 1 above)
Comment (nomis52) [inline firmware/src/system_config/number8/board_init.h]: Done

File: firmware/src/usb_descriptors.c (hunk 2 above)
Comment (nomis52) [inline firmware/src/usb_descriptors.c]: Removed the scaling comment

File: tests/system_config/bootloader_settings.h (hunk 3 above)
Comment (nomis52) [inline tests/system_config/bootloader_settings.h]: Oh that was the old style, I've removed the lines above

File: firmware/src/system_config/ethernet_sk2/board_init.c [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * board_init.c
```
Comment (peternewman) [inline firmware/src/system_config/ethernet_sk2/board_init.c]: Want to add a folder to this one too, just to finish the job.

**Comment by peternewman** [comment]
Just one more comment, then merge away.

File: firmware/src/system_config/ethernet_sk2/board_init.c (hunk 4 above)
Comment (nomis52) [inline firmware/src/system_config/ethernet_sk2/board_init.c]: Done

**Comment by peternewman** [comment]
LGTM

**Pull request merged**